packages/ui/src/app/pages/Election/components/ElectionProgressBar.tsx

```diff
@@ -0,0 +1,459 @@
+import BN from 'bn.js'
+import { sum } from 'lodash'
+import React, { useState, useEffect, useMemo } from 'react'
+import ReactDOM from 'react-dom'
+import { usePopper } from 'react-popper'
+import styled from 'styled-components'
+
+import { ProgressBar, ProgressBarProps } from '@/common/components/Progress'
+import {
+  MultiStatisticItem,
+  StatisticItemSpacedContent,
+  StatisticLabel,
+  StatisticItemProps,
+} from '@/common/components/statistics'
+import { TooltipText } from '@/common/components/Tooltip'
+import { DurationValue } from '@/common/components/typography/DurationValue'
+import { AN_HOUR, A_DAY, A_MINUTE, BorderRad, Colors, Transitions, ZIndex } from '@/common/constants'
+import { splitDuration, MILLISECONDS_PER_BLOCK } from '@/common/model/formatters'
+import { toNumber } from '@/common/utils'
+import { useCouncilConstants } from '@/council/hooks/useCouncilConstants'
+
+interface ElectionProgressBarProps extends StatisticItemProps {
+  electionStage: string
+  value?: number | BN
+  currentBlock?: number | BN
+}
+
+export const ElectionProgressBar = (props: ElectionProgressBarProps) => {
+  const currentBlock = toNumber(props.currentBlock)
+  const duration = toNumber(props.value)
+
+  const convertDurationToTimeString = (duration: number) => {
+    const format = splitDuration([
+      [A_DAY, 'd'],
+      [AN_HOUR, 'h'],
+      [A_MINUTE, 'm'],
+    ])
+
+    const formatDurationDate = (duration: number): [string | number, string][] => {
+      if (duration * MILLISECONDS_PER_BLOCK < A_MINUTE) {
+        return [['< 1', 'm']]
+      }
+      return format(duration * MILLISECONDS_PER_BLOCK)
+    }
+
+    return <DurationValue value={formatDurationDate(duration)} />
+  }
```

Comment:
Please also move `convertDurationToTimeString` out of the component context but also please make it into a component:
```tsx
const Duration = ({ duration }: { duration: number }) => {
```